### PR TITLE
Proxy test

### PR DESCRIPTION
**Pull request by echen94**
- added unit tests for proxy handler (with no dependency injection yet)
- also added some helper functions in some files
- inside proxy handler handle request function, it seems like it would dump out gibberish on the terminal when it tries to connect to the website

### CODE REVIEW

File: proxy_handler.cc [hunk 1]
```diff
@@ -28,10 +28,10 @@ RequestHandler::Status ProxyHandler::Init(const std::string& uri_prefix, const N
 		if(remote_host_found == true)
 		    return RequestHandler::Status::OK;
```
Comment (yanzunh) [inline proxy_handler.cc]: may add brackets to this line as well

File: proxy_handler.cc [hunk 2]
```diff
@@ -205,7 +205,7 @@ RequestHandler::Status ProxyHandler::HandleRequest(const Request& request, Respo
 
         
     }
-
+	std::cout<<"out of loop";
```
Comment (yanzunh) [inline proxy_handler.cc]: it might be better if we remove the debug logging message

File: proxy_handler_test.cc [hunk 3]
```diff
@@ -1,68 +1,94 @@
...
+			auto init_status = proxy_handler.Init("/", out_config);
+			EXPECT_EQ(init_status, RequestHandler::Status::NOT_OK);
+		}
+		//illegal config with too many statements
+		TEST(ProxyHandler, TooManyStatements)
+		{
+			NginxConfigParser parser;
+			NginxConfig out_config;
+			ProxyHandler proxy_handler;
+            std::stringstream config_stream("remote_host http://www.ucla.edu;remote_host2 http://www.google.com");
```
Comment (yanzunh) [inline proxy_handler_test.cc]: there might be an indentation mistake here. Also, an alternative to declaring same variables in every test case is to pull them out and put them into a class

File: proxy_handler_test.cc [hunk 4]
```diff
@@ -1,68 +1,94 @@
...
+			EXPECT_EQ(init_status, RequestHandler::Status::OK);
+
+			auto status = proxy_handler.HandleRequest(*req, &response);
+			EXPECT_EQ(status, RequestHandler::Status::OK);
+			
+		}
+
+
+   }//server
+}//http
```
Comment (yanzunh) [inline proxy_handler_test.cc]: Nice testcases!!!!

**Review by yanzunh — COMMENTED** [review]
In general very nice tests. Just a couple recommendations concerning coding styles

**Comment by neosbob** [comment]
The gibberish is actually the body of the response from the remote host, I printed it out to see if proxy is receiving the right responses from the remote server.  The test cases look like they are good to go, just need to do some dependency injection.

**Comment by echen94** [comment]
Thanks for the comments on the styles! definitely need to be more careful with indentations, cout in source files, and missing of brackets for one line if statements. Will make changes and do pull request again.

**Pull request merged**